File: frontend/src/types/ambulance.ts
export type AmbulanceStatus = 'AVAILABLE' | 'ASSIGNED' | 'EN_ROUTE' | 'ARRIVED' | 'OFFLINE';

export interface Ambulance {
  id: number;
  vehicle_number: string;
  status: AmbulanceStatus;
  current_latitude?: number;
  current_longitude?: number;
  created_at: string;
  updated_at: string;
}

export interface AmbulanceAssignRequest {
  ambulance_id: number;
}

export interface AmbulanceStatusUpdate {
  status: AmbulanceStatus;
}

File: backend/app/schemas/ambulance.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict
from app.models.ambulance import AmbulanceStatus

class AmbulanceBase(BaseModel):
    vehicle_number: str
    status: AmbulanceStatus = AmbulanceStatus.AVAILABLE
    current_latitude: Optional[float] = None
    current_longitude: Optional[float] = None

class AmbulanceCreate(AmbulanceBase):
    pass

class AmbulanceRead(AmbulanceBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class AmbulanceAssignRequest(BaseModel):
    ambulance_id: int

class AmbulanceStatusUpdate(BaseModel):
    status: AmbulanceStatus
